### PR TITLE
Use package.yaml in GUIDE.md

### LINKED ISSUE #3755: Update documentation to use package.yaml
From https://github.com/commercialhaskell/stack/issues/3633#issuecomment-355841756

> FYI the section on [Listing Dependencies](https://github.com/commercialhaskell/stack/blob/98c51cafb038bf09b58d5607956fb199d3660735/doc/GUIDE.md#listing-dependencies) still needs updating to reference `package.yaml` instead of the `.cabal`, > i.e: > https://github.com/commercialhaskell/stack/commit/6a091f8dfc7e2d821a2c065517833ecbc5fd0285#diff-4f532e9d254305a02dab50da05285712R318

Also worthwhile to skim the rest of the docs and see if anything needs updating.

### PR DESCRIPTION
**Pull request by dukedave**
Closes #3755

### CODE REVIEW

File: doc/GUIDE.md [hunk 1]
```diff
@@ -340,7 +340,7 @@ does so automatically.
 ### Listing Dependencies
 
 Let's have stack add a few more dependencies to our project. First, we'll include two new packages in the
-`build-depends` section for our library in our `helloworld.cabal`:
+`build-depends` section for our library in our `package.yaml``:
```
Comment (kadoban) [inline doc/GUIDE.md]: Looks like an extra backtick snuck in after `package.yaml`, unless it's just the diff looking funny.

Comment (dukedave) [reply doc/GUIDE.md]: Good :eyes:, amended.
Also actually checked [here](https://github.com/dukedave/stack/blob/use-hpack-in-guide/doc/GUIDE.md) and LGTM.

File: doc/GUIDE.md [hunk 2]
```diff
@@ -340,7 +344,7 @@ does so automatically.
 ### Listing Dependencies
 
 Let's have stack add a few more dependencies to our project. First, we'll include two new packages in the
-`build-depends` section for our library in our `helloworld.cabal`:
+`build-depends` section for our library in our `package.yaml`:
```
Comment (sol) [inline doc/GUIDE.md]: This should be `dependencies` instead of `build-depends` I guess.

File: doc/GUIDE.md [hunk 3]
```diff
@@ -192,9 +192,15 @@ test/Spec.hs
...
-`helloworld.cabal`, and `stack.yaml`.
+dwell on them here).
+
+The `LICENSE` file and `README.md` have no impact on the
+build.
+
+The `helloworld.cabal` file is updated automatically as part of the
+`stack build` process and should not be modified.
+
+The files of interest here are `Setup.hs`, `package.yaml`, , and `stack.yaml`.
```
Comment (sol) [inline doc/GUIDE.md]: `, , and `, I guess there is at least one comma too much here

**Comment by dukedave** [comment]
Thanks @sol, I pushed a rebase with your fixes.
I think I need my :eyeglasses: checked!

**Comment by mgsloan** [comment]
LGTM, thanks!

**Pull request merged**